### PR TITLE
Fix/abi spec v0.6.0 breaking change

### LINKED ISSUE #74: Incompatibility with Solidity 0.6 ABI's
0.6 ABI's are slightly different from 0.5. For example, methods in 0.6 no longer have `constant` or `payable` properties in the JSON.

One consequence is that `useCacheSend` hooks don't work (resulting in errors at run-time: `f.cacheSend is not a function`)

[Initial thread on Spectrum](https://spectrum.chat/trufflesuite/drizzle/hi-all-has-anyone-gotten-drizzle-to-play-nicely-with-solc-0-6~4f025de2-e4a1-4da3-9a83-ddecff5a2524)

### PR DESCRIPTION
**Pull request by ghost**
This PR fixes #74 . 

These changes allow drizzle to properly distinguish between functions that need to be bound to `cacheCall` vs `cacheSend`. Also included are unit tests and a separate UI test (see test-0.6.0-ABI under ui-tests) to verify that drizzle can consume 0.6.0+ ABI artifacts while preserving backwards compatibility with 0.5.0+ artifacts. The UI test uses a solc 0.6.0+ contract that incorporates both `payable` and `nonpayable` functions.

To run the unit tests: 
1. In root dir: `yarn`
2. Run unit tests: `yarn test`

To run the UI test:
1. In root dir: `yarn`
2. In another terminal, deploy a local blockchain: `yarn ganache` 
3. Change dir to the truffleV6 project: `cd ui-tests/truffleV6`
4. Compile and migrate the 0.6.0 contract: `yarn ganache:deploy`
5.  Change dir to ui test: `../test-0.6.0-ABI`
6. Launch the ui test `yarn serve:ui`

CC: @nicholaspai -  please review and let us know if there are any changes or missing items. Thank you!

Co-authored by @MalloryLeeWong 
Thanks @cds-amal for your feedback and guidance.

### CODE REVIEW

File: packages/store/src/DrizzleContract.js [hunk 1]
```diff
@@ -22,20 +22,18 @@ class DrizzleContract {
 
     for (var i = 0; i < this.abi.length; i++) {
       var item = this.abi[i]
-
-      if (item.type == 'function' && item.constant === true) {
+      if (isGetterFunction(item)) {
```
Comment (nicholaspai) [inline packages/store/src/DrizzleContract.js:25]: +1 for moving this logic to its own module

File: ui-tests/truffle/migrations/2_deploy_contracts.js [hunk 2]
```diff
@@ -1,9 +1,11 @@
 const SimpleStorage = artifacts.require("SimpleStorage");
 const TutorialToken = artifacts.require("TutorialToken");
 const ComplexStorage = artifacts.require("ComplexStorage");
+const Grue = artifacts.require("Grue");
```
Comment (cds-amal) [inline ui-tests/truffle/migrations/2_deploy_contracts.js]: Are these contracts used?

Comment (ghost) [reply ui-tests/truffle/migrations/2_deploy_contracts.js]: The first three contracts are used but we can remove Grue from the migrations and the contracts directory.

**Comment by ikropotov** [comment]
@xMNG @cds-amal @nicholaspai 
Is it going to develop/master/release any time soon (or #76)?

**Comment by cds-amal** [comment]
@ikropotov Apologies for the delay. Things have been hectic for the last few weeks. I've moved this as ready for review so we can get it out the door.

**Comment by DiscRisknBisque** [comment]
Tested on my local and it looks good to me. Thanks for your hard work @xMNG and @MalloryLeeWong! Looking forward to any future contributions! Again apologies for the delay in getting this looked at and merged.

**Pull request merged**